python/lsst/ts/standardscripts/maintel/m2/check_actuators.py

```diff
@@ -0,0 +1,262 @@
+
+__all__ = ["CheckActuators"]
+
+
+import asyncio
+import time
+
+import numpy as np
+import yaml
+from lsst.ts.idl.enums.Script import ScriptState
+from lsst.ts.observatory.control.maintel.mtcs import MTCS, MTCSUsages
+
+from ...base_block_script import BaseBlockScript
+
+# TODO: move constants from lsst.ts.m2com to ts-xml
+NUM_ACTUATOR, NUM_TANGENT_LINK = 78, 6
+
+
+class CheckActuators(BaseBlockScript):
+    """Perform a M2 bump test on either a selection of individual
+    actuators or on all axial actuators.
+
+    Parameters
+    ----------
+    index : `int`
+        Index of Script SAL component.
+
+    Notes
+    -----
+
+    **Checkpoints**
+
+    - "Running bump test on FA ID: {id}.": Check individual actuator.
+    - "M2 bump test completed.": Check complete.
+
+    """
+
+    def __init__(self, index, add_remotes: bool = True):
+        super().__init__(index=index, descr="Bump Test on M2 Axial Actuators")
+
+        self.mtcs = MTCS(
+            domain=self.domain,
+            intended_usage=None if add_remotes else MTCSUsages.DryTest,
+            log=self.log,
+        )
+
+        # Average duration (seconds) of a bump test on a single actuator
+        self.time_one_bump = 120
+
+        # default period of bump test in seconds
+        self.period = 60
+
+        # default +/- force to apply during bump test in N
+        self.force = 10
+
+    async def assert_feasibility(self):
+        """Verify that the system is in a feasible state before
+        running bump test.
+
+
+        """
+
+        for comp in self.mtcs.components_attr:
+            if comp != "mtm2":
+                self.log.debug(f"Ignoring component {comp}.")
+                setattr(self.mtcs.check, comp, False)
+
+        # Check all enabled and liveliness
+        await asyncio.gather(
+            self.mtcs.assert_all_enabled(),
+            self.mtcs.assert_liveliness(),
+        )
+
+    @classmethod
+    def get_schema(cls):
+        # not sure how to do this
+        # m2_actuator_ids_str = ",".join(
+        #     [str(actuator_id) for actuator_id in self.m2_axial_actuator_ids]
+        # )
+        m2_actuator_ids_str = ",".join(
+            [str(actuator_id) for actuator_id in np.arange(73)]
+        )
+
+        url = "https://github.com/lsst-ts/"
+        path = (
+            "ts_externalscripts/blob/main/python/lsst/ts/standardscripts/"
+            "maintel/m2/check_actuators.py"
+        )
+        schema_yaml = f"""
+        $schema: http://json-schema.org/draft-07/schema#
+        $id: {url}{path}
+        title: CheckAcutators v1
+        description: Configuration for Maintel M2 bump test SAL Script.
+        type: object
+        properties:
+            actuators:
+                description: Actuators to run the bump test.
+                oneOf:
+                  - type: array
+                    items:
+                      type: number
+                      enum: [{m2_actuator_ids_str}]
```

Comment:
Since this is a number, you could use min/max instead of doing it this way. Just replace `enum` with

```
minimum: 0
maximum: {NUM_ACTUATOR - NUM_TANGENT_LINK}
```